apps/dashboard/src/contract-ui/tabs/listings/components/listing-stats.tsx

```diff
@@ -1,87 +1,90 @@
 import { Skeleton, Stack, Stat, StatLabel, StatNumber } from "@chakra-ui/react";
-import {
-  useDirectListingsCount,
-  useEnglishAuctionsCount,
-} from "@thirdweb-dev/react";
-import type { MarketplaceV3 } from "@thirdweb-dev/sdk";
-import { BigNumber } from "ethers";
 import { useMemo } from "react";
+import type { ThirdwebContract } from "thirdweb";
+import { totalAuctions, totalListings } from "thirdweb/extensions/marketplace";
+import { useReadContract } from "thirdweb/react";
 import { Card } from "tw-components";
 
-interface ListingStatsV3Props {
-  contract?: MarketplaceV3;
-  features: string[];
-}
-
-const TotalListingsStat: React.FC<{ contract?: MarketplaceV3 }> = ({
+const TotalListingsStat: React.FC<{ contract: ThirdwebContract }> = ({
   contract,
 }) => {
-  const directListingsQuery = useDirectListingsCount(contract);
-  const englishAuctionsQuery = useEnglishAuctionsCount(contract);
-
-  const totalListings = useMemo(
-    () =>
-      BigNumber.from(directListingsQuery?.data || 0).add(
-        BigNumber.from(englishAuctionsQuery?.data || 0),
-      ),
-    [directListingsQuery?.data, englishAuctionsQuery?.data],
+  const { data: directListingCount, isSuccess: listingLoaded } =
+    useReadContract(totalListings, {
+      contract,
+    });
+  const { data: englishAuctionCount, isSuccess: auctionLoaded } =
+    useReadContract(totalAuctions, {
+      contract,
+    });
+  const _totalListings = useMemo(
+    () => (directListingCount || 0n) + (englishAuctionCount || 0n),
+    [directListingCount, englishAuctionCount],
   );
 
   return (
     <Card as={Stat}>
       <StatLabel mb={{ base: 1, md: 0 }}>Total Listings</StatLabel>
-      <Skeleton
-        isLoaded={
-          !contract ||
-          (directListingsQuery.isSuccess && englishAuctionsQuery.isSuccess)
-        }
-      >
-        <StatNumber>{totalListings.toString()}</StatNumber>
+      <Skeleton isLoaded={listingLoaded && auctionLoaded}>
```

Comment:
now u can do `x.loading` and `y.loading` here which means it easier to follow the data flow